src/connectors/current-refinements/connectCurrentRefinements.js

```diff
@@ -290,19 +290,19 @@ function compareObjects(a, b, attribute) {
 
 function groupItemsByRefinements(items, helper) {
   return items.reduce(
-    (results, currentItem) =>
-      [
-        {
-          attribute: currentItem.attribute,
-          refinements: items
-            .filter(result => result.attribute === currentItem.attribute)
-            .sort((a, b) =>
-              compareObjects(a, b, a.type === 'numeric' ? 'value' : 'label')
-            ),
-          refine: refinement => clearRefinement(helper, refinement),
-        },
-        ...results.filter(result => result.attribute !== currentItem.attribute),
-      ].sort((a, b) => compareObjects(a, b, 'attribute')),
+    (results, currentItem) => [
+      ...results.filter(result => result.attribute !== currentItem.attribute),
+      {
+        attribute: currentItem.attribute,
+        refinements: items
+          .filter(result => result.attribute === currentItem.attribute)
+          // We want to keep the order of refinements except the numeric ones.
+          .sort(
+            (a, b) => (a.type === 'numeric' ? compareObjects(a, b, 'value') : 0)
```

Comment:
looks like it can be

Reply:
ok. I assumed the engine would only return real numbers.

Reply:
Indeed, this would work:

```js
sort((a, b) => (a.type === 'numeric' ? a.value - b.value : 0)),
```

The value is a number because it's converted beforehand: https://github.com/algolia/instantsearch.js/blob/011d47909894837f2c2da5088bc81803135d201b/src/connectors/current-refinements/connectCurrentRefinements.js#L261

I can therefore remove the `compareObjects()` function.